src/components/FilterBar/FilterDisplay/Inputs/SelectInput.react.js

```diff
@@ -1,8 +1,16 @@
 export class SelectInput extends React.Component {
-  constructor(props) {
-    super(props);
+  constructor(props, context) {
```

Comment:
Remember that React only calls the constructor once, even when it's reusing the component. Is this the behaviour you want?